### ISSUE TITLE
Using multisig for vote signatures

### ISSUE THREAD

**Issue by kaar2** [semuxproject/semux#738]
According to #131 the size of the vote signatures alone grows by 10 GB a year. Right now (after #715 has been implemented) each block contains 67 votes in the worst case. Using multisig can dramatically reduce the space of the voting signatures in the block as it's size is the same as a single signature. That would result in ~85% space reduction. I am not a crypto expert so I don't know if it's feasible to implement such algorithm in the consensus, but I think this is something that's worth considering.

**Comment by cryptokat** [comment]
@kaar2 Thank you for the proposal.
However I don't think this proposal is feasible since that would require exposing validators' private keys to each other if I understand your idea correctly.

**Comment by kaar2** [comment]
Not necessarily, there are ways to achieve that without sharing the private keys, for example:
https://tools.ietf.org/id/draft-ford-cfrg-cosi-00.html

**Comment by cryptokat** [comment]
@kaar2 Thanks for the link! We will look into how Cosi can be implemented in SemuxBFT.

**Comment by cryptokat** [comment]
There is an implementation of CoSi in golang for reference: https://github.com/dedis/cothority/tree/master/cosi

**Comment by AlexSemuxTrader** [comment]
Size of Semux DB is 1.5Gb. It too large for ~88000 transactions.

**Comment by phash** [comment]
https://github.com/dedis/cothority/tree/master/ftcosi
It's updated

**Comment by AlexSemuxTrader** [comment]
It's necessary to make reducing of blockchain ASAP.
